simtools/layout/layout_array.py

```diff
@@ -117,15 +133,84 @@ def _initialize_corsika_telescope(self, corsika_dict=None):
 
         if corsika_dict is not None:
             self._logger.debug(f"Initialize CORSIKA telescope parameters from dict: {corsika_dict}")
-            self._initialize_corsika_telescope_from_dict(corsika_dict)
         else:
             self._logger.debug("Initialize CORSIKA telescope parameters from file")
-            self._initialize_corsika_telescope_from_dict(
-                collect_data_from_yaml_or_dict(
-                    self.io_handler.get_input_data_file("parameters", "corsika_parameters.yml"),
-                    None,
-                )
+            corsika_dict = self._from_corsika_file_to_dict()
+
+        self._initialize_corsika_telescope_from_dict(corsika_dict)
+
+    def _from_corsika_file_to_dict(self, file_name=None):
+        """
+        Get the corsika parameter file and return a dictionary with the keys necessary to\
+        initialize this class.
+
+        Parameters
+        ----------
+        file_name: str or Path
+            File from which to extract the corsika parameters. Default is \
+            data/parameters/corsika_parameters.yml
+
+        Returns
+        ------
+        corsika_dict:
+            Dictionary with corsika telescopes information.
+
+        Raises
+        ------
+        FileNotFoundError:
+            If file_name does not exist.
+        """
+        logger = logging.getLogger(__name__)
+        if file_name is None:
+            corsika_parameters_dict = collect_data_from_yaml_or_dict(
+                self.io_handler.get_input_data_file("parameters", "corsika_parameters.yml"), None
             )
+        else:
+            if not isinstance(file_name, Path):
+                file_name = Path(file_name)
+            if file_name.exists():
+                corsika_parameters_dict = collect_data_from_yaml_or_dict(file_name, None)
+            else:
+                raise FileNotFoundError
+
+        corsika_dict = {}
+        corsika_pars = ["corsika_sphere_radius", "corsika_sphere_center"]
```

Comment:
side remark: this should go away in future, as both parameters are not really related to layouts (and partly dependent on the structure)

(but now ok!)

Reply:
That is right. We want to read all this from the database, which will have separated site and model parameters.